adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java

```diff
@@ -936,22 +936,40 @@ public final void uploadCommandResponseMessage(
                     .withTag(TracingHelper.TAG_AUTHENTICATED.getKey(), authenticatedDevice != null)
                     .start();
 
-            // send response message to application via sender link
-            sendCommandResponse(tenant, commandResponse, currentSpan.context())
-            .map(delivery -> {
-                metrics.incrementCommandResponseDeliveredToApplication(tenant);
-                LOG.trace("delivered command response [command-request-id: {}] to application", commandRequestId);
-                currentSpan.log("delivered command response to application");
-                ctx.response().setStatusCode(HttpURLConnection.HTTP_ACCEPTED);
-                ctx.response().end();
-                return delivery;
+            final Future<TenantObject> tenantConfigTracker = getTenantConfiguration(tenant, currentSpan.context());
+            tenantConfigTracker.compose(tenantObj -> {
```

Comment:
In contrast to the `doUploadMessage` method we do not need the `tenantConfigTracker` for further handling,
so I propose to just code it like 

`getTenantConfiguration(tenant, ...).compose(tenantObj -> {
. . . 
});`